src/Runner.Listener/SelfUpdater.cs

```diff
@@ -220,12 +222,25 @@ private async Task DownloadLatestRunner(CancellationToken token)
 
                                 Trace.Info($"Downloading {_targetPackage.DownloadUrl}");
 
-                                using (FileStream fs = new FileStream(archiveFile, FileMode.Create, FileAccess.Write, FileShare.None, bufferSize: 4096, useAsync: true))
+                                using (FileStream fs = new FileStream(archiveFile, FileMode.Create, FileAccess.ReadWrite, FileShare.None, bufferSize: 4096, useAsync: true))
                                 using (Stream result = await httpClient.GetStreamAsync(_targetPackage.DownloadUrl))
                                 {
                                     //81920 is the default used by System.IO.Stream.CopyTo and is under the large object heap threshold (85k). 
                                     await result.CopyToAsync(fs, 81920, downloadCts.Token);
                                     await fs.FlushAsync(downloadCts.Token);
+
+                                    // Validate Hash Matches if it is provided
+                                    if (!String.IsNullOrEmpty(_targetPackage.HashValue))
+                                    {
+                                        fs.Seek(0, SeekOrigin.Begin);
```

Comment:
I would suggest don't seek the stream back to 0, just read the after the download finish to make the code less indented with `using...using..using..if...if...`